components/Input/NumericInput.tsx

```diff
@@ -11,46 +11,45 @@ type NumericInputProps = {
 	defaultValue?: any;
 	disabled?: boolean;
 	id?: string;
+	bold?: boolean;
 };
 
 const INVALID_CHARS = ['-', '+', 'e'];
 
-const NumericInput: FC<NumericInputProps> = memo(
-	({ value, onChange, placeholder, className, defaultValue, disabled, id, ...rest }) => {
-		const handleOnChange = useCallback(
-			(e: ChangeEvent<HTMLInputElement>) => {
-				const { value } = e.target;
-
-				onChange(e, value.replace(/,/g, '.').replace(/[e+-]/gi, ''));
-			},
-			[onChange]
-		);
-
-		return (
-			<StyledInput
-				{...rest}
-				value={value}
-				type="number"
-				onChange={handleOnChange}
-				placeholder={placeholder}
-				className={className}
-				onKeyDown={(e) => {
-					if (INVALID_CHARS.includes(e.key)) {
-						e.preventDefault();
-					}
-				}}
-				min="0"
-				step="any"
-				defaultValue={defaultValue}
-				disabled={disabled}
-				id={id}
-			/>
-		);
-	}
-);
-
-export const StyledInput = styled(Input)`
+const NumericInput: FC<NumericInputProps> = memo(({ onChange, bold, ...props }) => {
```

Comment:
Not clear what the difference is between NumericInput and CustomNumericInput is, is it possible to reduce to one with some variant if need be?

Reply:
The structure of the components is fundamentally different. You can look closer at the styled components in both for context. One of them has a wrapper and the other doesn't. The actual input in the `CustomNumericInput` is transparent inside a div. On the other hand, the one in the `NumericInput` is just a regular styled input. The custom one is that way because we use it to add special properties to the input (like a max button or left/right labels). The structure of the components is fundamentally different. You can look closer at the styled components in both for context. I can make a note to merge them in a separate PR. Merging them is a little more tricky than it looks at first glance, plus there are some details to iron out once the refactored designs are out (like using ch units in the CSS).